plugins/modules/net_tools/dnsimple.py

```diff
@@ -201,113 +200,135 @@ def main():
-            module.exit_json(changed=False, result=[d['domain'] for d in domains])
+            domain_list = []
+            domains_pagination = client.domains.list_domains(account.id, per_page=pagination_per_page).pagination
+            for page in range(1, domains_pagination.total_pages + 1):
+                page_data = client.domains.list_domains(account.id, page=page, per_page=pagination_per_page).data
+                domain_list.extend(page_data)
+            module.exit_json(changed=False, result=[d.__dict__ for d in domain_list])
 
         # Domain & No record
         if domain and record is None and not record_ids:
-            domains = [d['domain'] for d in client.domains()]
-            if domain.isdigit():
-                dr = next((d for d in domains if d['id'] == int(domain)), None)
-            else:
-                dr = next((d for d in domains if d['name'] == domain), None)
+            try:
+                if domain.isdigit():
+                    dr = client.domains.get_domain(account.id, int(domain)).data
+                else:
+                    dr = client.domains.get_domain(account.id, str(domain)).data
+            # domain does not exist
+            except DNSimpleException:
+                dr = None
             if state == 'present':
                 if dr:
-                    module.exit_json(changed=False, result=dr)
+                    module.exit_json(changed=False, result=dr.__dict__)
                 else:
                     if module.check_mode:
                         module.exit_json(changed=True)
                     else:
-                        module.exit_json(changed=True, result=client.add_domain(domain)['domain'])
-
+                        response = client.domains.create_domain(account.id, domain)
+                        module.exit_json(changed=True, result=response.data.__dict__)
             # state is absent
             else:
                 if dr:
                     if not module.check_mode:
-                        client.delete(domain)
+                        client.domains.delete_domain(account.id, domain)
                     module.exit_json(changed=True)
                 else:
                     module.exit_json(changed=False)
 
         # need the not none check since record could be an empty string
         if domain and record is not None:
-            records = [r['record'] for r in client.records(str(domain), params={'name': record})]
+            dr = client.domains.get_domain(account.id, domain).data
+
+            records_list = []
+            records_pagination = client.zones.list_records(account.id, dr.name, filter={'name': record}, per_page=pagination_per_page).pagination
+            for page in range(1, records_pagination.total_pages + 1):
+                page_data = client.zones.list_records(account.id, dr.name, filter={'name': record}, page=page, per_page=pagination_per_page).data
+                records_list.extend(page_data)
```

Comment:
Do you think we can add a common method to return paginated records or paginated domains? So that we do not repeat code.

Reply:
Yes absolutely, I will fix this

Reply:
I implemented this here https://github.com/ansible-collections/community.general/pull/2946/files#diff-5d983a5b601bdc6628f8aa0911b85fe1c3a2c33038a0d35982b5fefce8c3a518R300